### PR TITLE
ci(workflow): add cache to workflows using actions/setup-node

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

**Review by Imranba23 — APPROVED** [review]

**Review by Imranba23 — APPROVED** [review]

**Review by eliasbuchwald — APPROVED** [review]

File: .github/workflows/validate-data.yaml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: Validate Data
 
 on:
-  push:
-  pull_request:
-
+  push: null
+  pull_request: null
```
Comment (andymckay) [inline .github/workflows/validate-data.yaml:5]:
```suggestion
  push:
  pull_request:
```

The `null` isn't needed and isn't in other starter workflows, or in our docs so let's remove that.

**Review by andymckay — COMMENTED** [review]
With the suggested changes, this looks good to go, thank you.

**Review by 017963355 — APPROVED** [review]

**Review by andymckay — APPROVED** [review]
Thank you @oscard0m

File: .github/workflows/validate-data.yaml [hunk 2]
```diff
@@ -13,6 +12,7 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: "12"
+          cache: npm
```
Comment (andymckay) [inline .github/workflows/validate-data.yaml:15]: And similarly here, this won't work.

**Comment by andymckay** [comment]
I'll note that the starter workflow script our users get is [here](https://github.com/actions/starter-workflows/blob/7135450ec14e7e47bf3e41dd65d399e5e315fa39/ci/node.js.yml) and has the `cache` value set. So I don't think we want this PR at all at this point, sorry.

**Comment by oscard0m** [comment]
> I'll note that the starter workflow script our users get is [here](https://github.com/actions/starter-workflows/blob/7135450ec14e7e47bf3e41dd65d399e5e315fa39/ci/node.js.yml) and has the `cache` value set. So I don't think we want this PR at all at this point, sorry.

Thanks for the feedback and your time reviewing this

**Pull request closed without merging**